lib/fastlane/plugin/wpmreleasetoolkit/helper/android_localize_helper.rb

```diff
@@ -169,3 +179,25 @@ def self.verify_pr_diff(main, library, main_strings, lib_strings, source_diff)
     end
   end
 end
+
+class Nokogiri::XML::Node
+  # Return true if this node is content-equivalent to other, false otherwise
+  def =~(other)
+    return true if self == other
+    return false unless name == other.name
+    stype = node_type; otype = other.node_type
+    return false unless stype == otype
+    sa = attributes; oa = other.attributes
+    return false unless sa.length == oa.length
+    sa = sa.sort.map{ |n,a| [n,a.value,a.namespace && a.namespace.href] }
+    oa = oa.sort.map{ |n,a| [n,a.value,a.namespace && a.namespace.href] }
+    return false unless sa == oa
+    skids = children; okids = other.children
+    return false unless skids.length == okids.length
+    return false if stype == TEXT_NODE && (content != other.content)
+    sns = namespace; ons = other.namespace
+    return false if !sns ^ !ons
+    return false if sns && (sns.href != ons.href)
+    skids.to_enum.with_index.all?{ |ski,i| ski =~ okids[i] }
+  end
+end
```

Comment:
I'm guessing this code comes from [this StackOverflow question & answer](https://stackoverflow.com/questions/7825258/determine-if-two-nokogiri-nodes-are-equivalent?rq=1), right?

The author mentions that at the time of writing Nokogiri didn't have support for canonicalization, but [now it does](https://nokogiri.org/rdoc/Nokogiri/XML/Document#canonicalize-instance_method). 

I would be good to add a link to the question so that if we'll ever have time (and understanding of Nogokiri) we can replace this confusing code with something simpler.

Reply:
It's probably code that has been around for a while as I found it in GitHub following a link from a blog post :-), but I agree that it's a good idea to reference and mention it.